tools/build-version.go

```diff
@@ -49,7 +49,7 @@ func main() {
 	}
 	// Get the tag of the current revision.
 	tag, _ := getTag("--exact-match")
-	if tag == versionStr {
+	if tag == versionStr || ahead == nil {
```

Comment:
But doesn't that mean we've no `versionStr` right now?
I know we're coming from #3514 resp. https://github.com/termux/termux-packages/issues/21286.

Reply:
If we do this if `ahead` is nil no matter what `getTag("--exact-match")` returned, why don't we check `ahead` before even calling `getTag("--exact-match")`?

Reply:
I didn't think much about checking `ahead` before calling `getTag("--exact-match")`. It does not have to be called when there are commits between the last version tag because the tag returned is different, so I will move the if statement before the call and remove `tag == versionStr`.